### PR TITLE
FEATURE: Use localDistributionFolders on ``package:create``

### PR DESCRIPTION
**Pull request by mficzel**
If the composer file has a local distribution folder identified by type "path" and
a local folder reference in the "url" new packages are created in the
distributionFolder and are required via 'composer require ... @dev'.

### CODE REVIEW

File: Neos.Flow/Classes/Command/PackageCommandController.php [hunk 1]
```diff
@@ -78,10 +78,11 @@ public function injectBootstrap(Bootstrap $bootstrap)
      * @Flow\FlushesCaches
      * @param string $packageKey The package key of the package to create
      * @param string $packageType The package type of the package to create
+     * @param string $packagesPath The path where new packages are created relative to the flow root path
      * @return void
      * @see neos.kickstarter:kickstart:package
      */
-    public function createCommand(string $packageKey, string $packageType = PackageInterface::DEFAULT_COMPOSER_TYPE)
+    public function createCommand(string $packageKey, string $packageType = PackageInterface::DEFAULT_COMPOSER_TYPE, ?string $packagesPath)
```
Comment (daniellienert) [inline Neos.Flow/Classes/Command/PackageCommandController.php]: What do you think about setting 'DistributionPackages' as default here and use it if it exists? Would be a better default.

File: Neos.Kickstarter/Classes/Command/KickstartCommandController.php [hunk 2]
```diff
@@ -46,10 +46,11 @@ class KickstartCommandController extends CommandController
      *
      * @param string $packageKey The package key, for example "MyCompany.MyPackageName"
      * @param string $packageType Optional package type, e.g. "neos-plugin"
+     * @param string $packagesPath The path where new packages are created relative to the flow root path.
      * @return string
      * @see neos.flow:package:create
      */
-    public function packageCommand($packageKey, $packageType = PackageInterface::DEFAULT_COMPOSER_TYPE)
+    public function packageCommand(string $packageKey, string $packageType = PackageInterface::DEFAULT_COMPOSER_TYPE, ?string $packagesPath)
```
Comment (daniellienert) [inline Neos.Kickstarter/Classes/Command/KickstartCommandController.php]:
```suggestion
    public function packageCommand(string $packageKey, string $packageType = PackageInterface::DEFAULT_COMPOSER_TYPE, ?string $packagesPath): void
```

File: Neos.Kickstarter/Classes/Command/KickstartCommandController.php [hunk 3]
```diff
@@ -46,10 +46,11 @@ class KickstartCommandController extends CommandController
      *
      * @param string $packageKey The package key, for example "MyCompany.MyPackageName"
      * @param string $packageType Optional package type, e.g. "neos-plugin"
+     * @param string $packagesPath The path where new packages are created relative to the flow root path.
      * @return string
```
Comment (daniellienert) [inline Neos.Kickstarter/Classes/Command/KickstartCommandController.php]:
```suggestion
     * @return void
```

File: Neos.Flow/Classes/Command/PackageCommandController.php (hunk 1 above)
Comment (mficzel) [reply Neos.Flow/Classes/Command/PackageCommandController.php]: I removed the change in this file and detect the DistributionPackages folder.

**Comment by kitsunet** [comment]
What could ever go wrong.. My bad should've thought this through, but you cannot run unit tests with an installation with distribution packages anymore as it will start composer when the unit test creates a package... I guess we need to fix this ASAP.

**Comment by mficzel** [comment]
@kitsunet is there an option to run the unit tests with a specified packagePath. Since the composer behavior is not wrong per se that should help aswell.

**Pull request merged**